dashboard/src/main/java/com/google/cloud/tools/opensource/cloudbomdashboard/VersionData.java

```diff
@@ -0,0 +1,278 @@
+    private final Map<String, String> updatedTime = new HashMap<>();
+    private final Map<String, String> metadataURL = new HashMap<>();
+
+    private static DateFormat dateFormat = new SimpleDateFormat("yyyyMMddhhmmss");
+    private static DateFormat outputFormat = new SimpleDateFormat("MM-dd-yyyy");
+
+    public VersionData(String cloudBomVersion) {
+        versions.add(cloudBomVersion);
+        ALL_VERSIONS_DATA.versions.add(cloudBomVersion);
+    }
+
+    //Constructor meant only for ALL_VERSIONS_DATA
+    private VersionData() {
+    }
+
+    /**
+     * Populate this dashboard with all artifacts in the keyset of the given mapping.
+     * If 'addToAllVersions' is set, we will also add every artifact to the 'All Versions' page,
+     * for creation in the
+     *
+     * @param addToAllVersions true to add all artifacts to the 'All Versions' page as well
+     * @param infoMap          Mapping of artifacts
+     */
+    public void populateData(boolean addToAllVersions, Map<Artifact, ArtifactInfo> infoMap) {
+        String cloudBomVersion = versions.get(0);
+        for (Map.Entry<Artifact, ArtifactInfo> info : infoMap.entrySet()) {
+            insertData(cloudBomVersion, info.getKey());
+
+            if (addToAllVersions) {
+                ALL_VERSIONS_DATA.insertData(cloudBomVersion, info.getKey());
+            }
+        }
+    }
+
+    /**
+     * Inserts an artifact's data into this VersionData.
+     * Pass cloudBomVersion for the purpose of our artifact key, used in our FTL file.
+     */
+    private void insertData(String cloudBomVersion, Artifact a) {
+        String artifactId = a.getArtifactId();
+        //Concatenate this with the version of the current cloud BOM, so each entry has a unique
+        //key for the mapping in our table.
+        String artifactKey = artifactId + ":" + cloudBomVersion;
+        String groupId = a.getGroupId();
+        String version = a.getVersion();
+
+        String latestVersion = latestVersion(a);
+        String pomFileURL = getPomFileURL(groupId, artifactId, version);
+        String sharedDependencyVersion = sharedDependencyVersion(artifactKey, a, sharedDependenciesPosition);
+
+        artifacts.add(artifactId);
+        currentVersion.put(artifactKey, version);
+        newestVersion.put(artifactKey, latestVersion);
+        newestPomURL.put(artifactKey, pomFileURL);
+        sharedDepsVersion.put(artifactKey, sharedDependencyVersion);
+        updatedTime.put(artifactKey, updatedTime(a));
+        metadataURL.put(artifactKey, getMetadataURL(a));
+    }
+
+    /**
+     * Creates new mapping of template data for use in creating the All Versions page.
+     */
+    public static Map<String, Object> getAllVersionsTemplate() {
+        return ALL_VERSIONS_DATA.getTemplateData();
+    }
+
+    /**
+     * Returns the 'template data' formatting of our data.
+     */
+    public Map<String, Object> getTemplateData() {
+        Map<String, Object> templateData = new HashMap<>();
```

Comment:
If you need maps for this--I'm not sure you do--then they can be created here rather than being instance fields.